docs/gameplay-components/custom-settings.md

```diff
@@ -0,0 +1,72 @@
+# Configuring Custom Settings
+
+## How do I add custom settings to my game?
+
+The settings config for the game can be found in `src/main.js`.
+
+```javascript
+const settingsConfig = {
+    pages: [
+        {
+            title: "Global Settings",
+            settings: [
+                {
+                    key: "audio",
+                    type: "toggle",
+                    title: "Audio",
+                    description: "Turn off/on sound and music",
+                },
+                {
+                    key: "custom1",
+                    type: "toggle",
+                    title: "Custom 1",
+                    description: "Switch custom message",
+                },
+            ],
+        },
+    ],
+};
+```
+
+A new setting can be created by adding an additional object to the `settings` array and giving it the following properties:
+- `key`: A key value used to identify the setting.
+- `type`: Determines the format of the setting. Only `"toggle"` is supported currently.
+- `title`:The name of the setting as it will appear in the settings dialog box.
+- `description`: A short description of what the setting does. This will appear under the settings title in the settings dialog box.
+
+## How do I access the value of a setting within the gameplay component?
+Import Genie's settings module and then call its `getAllSettings()` function. Under the hood this will retrieve the settings from the GMI.
+
+Example usage:
+
+```javascript
+import { settings } from "../node_modules/genie/src/core/settings.js";
+
+const allSettings = settings.getAllSettings();
+// => { audio: true, custom1: false, gameData: {} }
+
+allSettings.audio;
+// => true
+
+allSettings.custom1;
+// => false
+```
+
+## How do I configure a callback for my setting?
+
+In `src/main.js` subscribe to the settings channel by calling `signal.bus.subscribe(subscription)` where `subscription` is an object with the following properties:
+- `channel`: This should be set to settingsChannel which is imported at the top of `src/main.js`.
+- `name`: This should correspond to the key of the setting that the callback is being set for.
+- `callback`: A callback function that will execute whenever the setting is changed from the settings dialog box.
+
+Example usage:
+
+```javascript
+signal.bus.subscribe({
+    channel: settingsChannel,
```

Comment:
I think I've already made brief mention of this on Friday to some people but if settings should always use the settings channel we should probably do a small bit of refactoring at some point so we don't need agencies to add this. Maybe we should just have some function they can call like `addSettingsCallback("settingName", () => { do stuff });`

Reply:
Funny you should say this, Sean and I were just talking about it!

Reply:
Great minds and all that. 😆

Reply:
Sounds good to me. Would be best being another ticket 👍

Reply:
Yeah, I agree. We'll have to remember to update these docs as a part of it though.